ICSharpCode.Decompiler/IL/Transforms/NullPropagationTransform.cs

```diff
@@ -319,6 +319,218 @@ public void Run(Block block, int pos, StatementTransformContext context)
 			if (!context.Settings.NullPropagation)
 				return;
 			new NullPropagationTransform(context).RunStatements(block, pos);
+			if (TransformNullPropagationOnUnconstrainedGenericExpression(block, pos, out var target, out var value, out var rewrapPoint, out var endBlock)) {
+				context.Step("TransformNullPropagationOnUnconstrainedGenericExpression", block);
+				// A successful match was found:
+				// 1. The 'target' instruction, that is, the instruction where the actual 'null-propagating' call happens:
+				// <target>.Call() is replaced with <value>?.Call()
+				// 2. The 'value' instruction, that is, the instruction that produces the value:
+				// It is inlined at the location of 'target'.
+				// 3. The 'rewrapPoint' instruction is an ancestor of the call that is used as location for the NullableRewrap instruction,
+				// if the expression does not yet contain a NullableRewrap.
+				
+				// First try to find a NullableRewrap instruction
+				bool needsRewrap = true;
+				var tmp = target;
+				while (needsRewrap && tmp != null) {
+					if (tmp is NullableRewrap) {
+						needsRewrap = false;
+						break;
+					}
+					tmp = tmp.Parent;
```

Comment:
This loop seems weird.
Why would an arbitrary `NullableRewrap` ancestor be an appropriate jump target?

To maintain the semantics, the implicit nullable.unwrap jump can only jump over instructions that also weren't executed in the original IL in case of a `null`. So the pattern documented on `TransformNullPropagationOnUnconstrainedGenericExpression` is correct, but this code seems to be doing something else.